src/s-das.tex

```diff
@@ -0,0 +1,209 @@
+  ({\tt suatp}) Register}
+\label{s:das:suatp}
+
+The {\tt suatp} CSR controls address translation in U-mode and shares the
+same format as the {\tt satp} CSR.  The MODE field in {\tt suatp} must be
+written with the same value as the MODE field in {\tt satp} by supervisors
+using disjoint addressing mode; values different from MODE in {\tt satp}
+are reserved for future extensions that permit user and supervisor address
+spaces to use different paging depths.  Supervisors that use unified
+addressing mode will implicitly set the MODE fields in both {\tt suatp} and
+{\tt satp} to the same value when they write to the {\tt satp} CSR.
+
+If the MODE field in {\tt suatp} contains an illegal value, all user
+translations will fault.  Implementations must permit the MODE fields in
+{\tt satp} and {\tt suatp} to contain the same value.  Future extension may
+standardize other permitted values.
+
+\begin{commentary}
+The choice of retaining CSR 0x180 as {\tt satp} was driven by concerns for
+backwards compatibility and hardware simplicity.  This choice permits
+hardware to always operate in disjoint addressing mode, while unified
+addressing mode is provided by also writing {\tt suatp} when {\tt satp} is
+written.  A supervisor using disjoint addressing mode will write {\tt satp}
+once and then write {\tt suatp} on every task switch.  A supervisor that
+does not support disjoint addressing mode will write {\tt satp} on every
+task switch.
+\end{commentary}
+
+\subsection{Supervisor Status ({\tt sstatus}) Register}
+\label{s:das:SSU-SLU}
+
+Two one-bit fields are added to {\tt sstatus} (and therefore also to {\tt
+  mstatus} and any other CSRs that contain {\tt sstatus}) to provide S-mode
+software with easy access to disjoint user address spaces.
+
+\begin{figure*}[h!]
+{\footnotesize
+  \begin{center}
+    \setlength{\tabcolsep}{4pt}
+    \begin{tabular}{cccc}
+      \\
+      \instbit{} &
+      \instbit{24} &
+      \instbit{23} &
+      \instbit{} \\
+      \hline
+      \multicolumn{1}{c|}{$\cdots$} &
+      \multicolumn{1}{c|}{{\tt SSU}} &
+      \multicolumn{1}{c|}{{\tt SLU}} &
+      \multicolumn{1}{c}{$\cdots$} \\
+      \hline
+      & 1 & 1 & \\
+    \end{tabular}
+  \end{center}
+}
+\vspace{-0.1in}
+\caption{Additional Fields in {\tt sstatus}}
+\label{s:das:sstatus}
+\end{figure*}
+
+Implementations supporting disjoint addressing mode must implement SLU and
+SSU.  Implementations not supporting disjoint addressing mode must hardwire
+these bits to zero, as the fields are \wpri\ otherwise.
+
+The SSU and SLU bits affect instructions using a specific major opcode:
+\begin{compactitem}
+\item SLU (Supervisor Load from User memory) affects LOAD instructions
+\item SSU (Supervisor Store to User memory) affects STORE instructions
+\end{compactitem}
+
+The SLU bit also affects LQ on RV128, even though LQ is encoded using the
+MISC-MEM major opcode because LOAD is full.
+
+The SSU and SLU bits are ignored if SPP is S.
```

Comment:
This is pretty ugly and is going to end up causing problems. If you're doing a split address space you really need explicit translated load/store instructions, otherwise you're just asking to have something somewhere not save/restore sstatus appropriately and start accessing the wrong place.